File: app/src/main/java/com/games/memorygame/ui/BaseViewHolder.java
package com.themobilecompany.memorygame.ui;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import butterknife.ButterKnife;

public class BaseViewHolder<T> extends RecyclerView.ViewHolder {

    private final View view;
    private T item;

    public BaseViewHolder(View itemView) {
        super(itemView);
        view = itemView;
        ButterKnife.bind(this, itemView);
    }

    protected Context getContext() {
        return view.getContext();
    }

    public void onBindView(T item) {
        this.item = item;
    }

    protected View getItemView() {
        return view;
    }
}

File: app/src/main/java/com/games/memorygame/ui/BaseViewHolder.kt
package com.games.memorygame.ui

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.View

import butterknife.ButterKnife

open class BaseViewHolder<T>(itemView: View?) : RecyclerView.ViewHolder(itemView) {
    private var item: T? = null


    protected val context: Context
        get() = itemView.getContext()

    init {
        ButterKnife.bind(this, this.itemView)
    }

    open fun onBindView(item: T) {
        this.item = item
    }
}